spec/code_objects/constants_spec.rb

```diff
@@ -66,7 +66,9 @@
   describe :BUILTIN_CLASSES do
     it "includes all base classes" do
       bad_names = []
-      YARD::CodeObjects::BUILTIN_CLASSES.each do |name|
+      builtin_classes = YARD::CodeObjects::BUILTIN_CLASSES
+      builtin_classes -= ["Bignum", "Fixnum"] if 0.class == Integer
```

Comment:
As far as I'm aware, any access of `::Bignum` or `::Fixnum` will emit a deprecation warning in Ruby 2.4 (as in the `val(name).is_a?(Class)` at line 73 of `constants_spec.rb` above). The new code only omits those two classes in situations where that would be the case (where `0.class == Integer`, which is only the case from 2.4 onwards).